compressor/tests/test_base.py

```diff
@@ -332,6 +335,52 @@ def test_get_cachekey_leading_dot(self):
     def test_get_cachekey_invalid_mod(self):
         self.assertRaises(ImportError, lambda: get_cachekey("foo"))
 
+
+@override_settings(
+    COMPRESS_ENABLED=True,
+    COMPRESS_PRECOMPILERS=(),
+    COMPRESS_DEBUG_TOGGLE='nocompress',
+    DEBUG=True
+)
+class CompressorInDebugModeTestCase(SimpleTestCase):
+
+    def setUp(self):
+        self.css = '<link rel="stylesheet" href="/static/css/one.css" type="text/css" />'
+        self.tmpdir = mkdtemp()
+        copytree(settings.STATIC_ROOT, os.path.join(self.tmpdir, "static"))
+        self.old_compress_root = settings.COMPRESS_ROOT
+        settings.STATIC_ROOT = settings.COMPRESS_ROOT = os.path.join(self.tmpdir, "static")
+        settings.STATICFILES_DIRS = [self.old_compress_root]
+
+    def tearDown(self):
+        rmtree(self.tmpdir)
+        settings.STATIC_ROOT = settings.COMPRESS_ROOT = self.old_compress_root
+        delattr(settings, "STATICFILES_DIRS")
```

Comment:
you mean this old_compress_root thing? that's done all over the place in django-compressor, and i don't know a better way to do it when the new setting value is not known before (so you can't use the decorator.

or do you mean the delattr thing? i don't know whether the original PR did that, it should be possible to save the original value with an old_* variable and restore it here, just like for the other settings